services/121-service/src/events/events.service.ts

```diff
@@ -254,25 +270,41 @@ export class EventsService {
     // Filter out the keys that are not in the registeredAttributes
     if (registeredAttributes) {
       fieldNames = fieldNames.filter((key) =>
-        registeredAttributes.includes(key),
+        registeredAttributes.includes(key as string),
       );
     }
 
     const events: EventEntity[] = [];
-    for (const fieldName of fieldNames) {
+    for (const key of fieldNames) {
       if (
-        oldEntity[fieldName] === newEntity[fieldName] ||
-        (isObject(oldEntity[fieldName]) &&
-          isObject(newEntity[fieldName]) &&
-          isMatch(oldEntity[fieldName], newEntity[fieldName]))
+        oldEntity[key] === newEntity[key] ||
+        (isObject(oldEntity[key]) &&
+          isObject(newEntity[key]) &&
+          isMatch(oldEntity[key], newEntity[key]))
       ) {
         continue;
       }
 
+      const oldValue =
+        typeof oldEntity[key] === 'string' || typeof oldEntity[key] === 'number'
+          ? oldEntity[key]
+          : oldEntity[key] && typeof oldEntity[key] === 'object'
+            ? (oldEntity[key] as LocalizedString)?.en ||
+              Object.values(oldEntity[key])[0]
+            : null;
+
+      const newValue =
+        typeof newEntity[key] === 'string' || typeof newEntity[key] === 'number'
+          ? newEntity[key]
+          : newEntity[key] && typeof newEntity[key] === 'object'
+            ? (newEntity[key] as LocalizedString)?.en ||
+              Object.values(newEntity[key])[0]
+            : null;
```

Comment:
I think this could use some more readable (and less repetitive) code...
Using a helper-function to write out the 'algorithm' to get to the final value.

Something like: (⚠️ not tested locally! And the exact typings used need a sanity-check)

```suggestion
      function getLocalizedStringOrFallback(value: LocalizedString | {}) {
        return (value as LocalizedString)?.en || Object.values(value)[0];
      }
      
      function getValue(value: unknown): string | number | null {
        if (typeof value === 'string' || typeof value === 'number') {
          return value;
        }
        
        if (isObject(value)) {
          return getLocalizedStringOrFallback(value);
        }
            
        return null;
      }
      
      const oldValue = getValue(oldEntity[key]);
      const newValue = getValue(newEntity[key]);
```

The helper-methods/functions could also be put somewhere more central/useful.. Probably not the only place where we do such tasks/magic. ;)

Reply:
Done